src/lib/rule2B.ts

```diff
@@ -0,0 +1,52 @@
+import {computeTextAlternative} from './compute_text_alternative';
+import {Context} from './context';
+
+/**
+ * Gets the valid idrefs listed in the aria-labelledby attribute of elem.
+ * @param elem - the element whose valid aria-labelledby idrefs are being
+ * calculated
+ * @return - a list of idref strings for whom elements exist in the document.
+ */
+function getValidAriaLabelledbyIdrefs(elem: HTMLElement): string[] {
+  let idrefs: string[] = [];
+  const idrefStr = elem.getAttribute('aria-labelledby');
+  if (idrefStr) {
+    idrefs = idrefStr.split(' ');
+  }
+  // A valid idref is considered here to mean an idref that identifies an
+  // existing element in the document.
+  const validIdrefs = idrefs.filter(idref => !!document.getElementById(idref));
+  return validIdrefs;
+}
+
+/**
+ * Implementation of rule 2B
+ * @param node - node whose text alternative is being computed
+ * @param context - Additional information relevant to the text alternative
+ * computation for node
+ * @return - The text alternative string is returned if condition is true,
+ * null is returned otherwise, indicating that the condition of this rule was
+ * not satisfied.
+ */
+export function rule2B(node: Node, context: Context): string | null {
+  let result = null;
+  if (node instanceof HTMLElement) {
```

Comment:
to make it more readbale, I would flatten the function to check one condition after another & then generate the whole label at once:

```typescript
if(!node instanceof HTMLElement) {
  return null;
}

if(context.ariaLabelledbyReference) {
  return null;
}

const labels = resolveValidAriaLabelledByIdRefs(node);
if(labels.length === 0){
  return null;
}
return labels.map(label => computeTextAlternative(label, {ariaLabelledbyReference: true})).join(' ');

```